guide/additional-info/changes-in-v13.md

```diff
@@ -0,0 +1,384 @@
+
+`MessageOptions.reply` (User ID) has been removed, replaced with `MessageOptions.replyTo` (Message ID)
+
+```diff
+- channel.send('content', { reply: '123456789012345678' })` // User ID
++ channel.send('content', { replyTo: '765432109876543219' })` // Message ID
+```
+
+The new `MessageOptions.allowedMentions.repliedUser` boolean option determines if the reply will notify the author of the original message.
+
+```diff
+- message.reply('content')
++ message.reply('content', { allowedMentions: { repliedUser: false }})
+```
+
+### Bitfields / Permissions
+
+The usage of string literals for bitfields such as `Permissions` and `UserFlags` is discouraged, you should use the flag instead.
+
+```diff
+- permissions.has('SEND_MESSAGES')
++ permissions.has(Permissions.FLAGS.SEND_MESSAGES)
+```
+
+### Webpack
+
+Webpack builds are no longer supported.
+
+## Changes and deletions
+
+### Client
+
+#### Client#emojis
+
+The Client Emoji manager is now a `BaseGuildEmojiManager`, providing cache resolution only and removing methods which would fail to create emojis as there was no Guild context.
+
+#### Client#generateInvite
+
+`Client#generateInvite` no longer supports `PermissionsResolvable` as its argument, requiring `InviteGenerationOptions`.
+
+To provide permissions, use `InviteGenerationOptions#permissions`.
+```diff
+- client.generateInvite([Permissions.FLAGS.SEND_MESSAGES]);
++ client.generateInvite({ permissions: [Permissions.FLAGS.SEND_MESSAGES] })
+```
+
+### ClientOptions
+
+#### ClientOptions#fetchAllMembers
+
+The `ClientOptions#fetchAllMembers` option has been removed.
+
+With the introduction of gateway intents, the `fetchAllMembers` Client option would often fail and causes major delays in ready states or even cause timeout errors. As its purpose is contradictory to Discord's intentions to reduce scraping of user and presence data, it has been removed.
+
+#### ClientOptions#messageEditHistoryMaxSize
+
+The `ClientOptions#messageEditHistoryMaxSize` option has been removed.
+
+To reduce on caching, discord.js will no longer store an edit history. This will need to be manually implemented if required.
+
+### Guild
+
+#### Guild#member
+
+The helper/shortcut method `Guild#member()` has been removed.
+
+```diff
+- guild.member(user);
++ guild.members.cache.get(user.id)
+```
+
+#### Guild#voice
+
+The `Guild#voice` getter has been removed.
+
+```diff
+- guild.voice
++ guild.me.voice
+```
+
+### GuildMember 
+
+#### GuildMember#ban
+
+`GuildMember#ban()` will throw a TypeError when a string is provided instead of an options object.
+
+```diff
+- member.ban('reason')
++ member.ban({ reason: 'reason' })
+```
+
+### GuildMember#hasPermission
+
+This shortcut method has been removed.
+
+```diff
+- member.hasPermission(Permissions.FLAGS.SEND_MESSAGES);
++ member.permissions.has(Permissions.FLAGS.SEND_MESSAGES);
```

Comment:
```suggestion
- member.hasPermission(Permissions.FLAGS.SEND_MESSAGES);
+ member.permissions.has(Permissions.FLAGS.SEND_MESSAGES) || member.id === guild.ownerID;
```

Reply:
That's already included in the other check: https://github.com/discordjs/discord.js/blob/master/src/structures/GuildMember.js#L219

Reply:
ah, so now `checkOwner` is always true? might want to include how to check if they have a permission without checking if they are owner (although idk if anyone actually set it to false)

Reply:
Has been the case for the GuildMember#permissions getter all along, but is neither documented nor optional. Seemingly an oversight of the author of the [PR](https://github.com/discordjs/discord.js/pull/5152) that removed #hasPermission:

>  Unlike the only addition checkOwner for this entire method, that the users can very easily implement.

This is wrong, as it's already implicitly included.

I personally don't really see a use case for these pseudo permissions, but if there is we could re-introduce the check in the #permissions getter.  In either case it should be documented.

Reply:
This isn't a permissions guide though. This is only supposed to be direct changes from v12 to v13